### ISSUE TITLE
 Service request: Soundcloud

### ISSUE THREAD

**Issue by fenryxo** [tiliado/nuvolaplayer#30]
From [Launchpad bug #1010111](https://bugs.launchpad.net/nuvola-player/+bug/1010111) by storminator:

Integrate Soundcloud http://soundcloud.com/

There are some patches for Nuvola Player 2 in the old bug report.

**Comment by florian-asche** [comment]
+1

**Comment by fenryxo** [comment]
Waiting for code review: https://github.com/AltarBeastiful/nuvola-app-soundcloud

**Comment by fenryxo** [comment]
Code review in progress: https://github.com/AltarBeastiful/nuvola-app-soundcloud/issues/1

**Comment by fenryxo** [comment]
Result of code review (AltarBeastiful/nuvola-app-soundcloud#1): Cannot merge in current state as the author haven't addressed my objections.

**Comment by fenryxo** [comment]
This ticket have been moved to the list of postponed tasks.

**Comment by fenryxo** [comment]
Currently blocked by legal issues and insufficient artist/track parsing. AltarBeastiful/nuvola-app-soundcloud#1

**Comment by fenryxo** [comment]
AltarBeastiful/nuvola-app-soundcloud#1 is a month without response, moving to postponed tasks.

**Comment by SteVwonder** [comment]
I am interesting in picking this up.  I will look into improving the parsing.

It looks like the code copyright has been fixed.  Is the logo the only legal issue remaining?

**Comment by fenryxo** [comment]
> I am interesting in picking this up. I will look into improving the parsing.

Great :+1: 

> It looks like the code copyright has been fixed. Is the logo the only legal issue remaining?

Yes, just remove the logo and I will ask Alexander King to create an icon also for Soundcloud :-)

**Comment by kixam** [comment]
hi @fenryxo and @SteVwonder ,

>  I will look into improving the parsing.

it seems I already did that a bit. Also, I added some extras, such as "Like" and "Shuffle" buttons.

after discussing with @SteVwonder at #161, I think he might be interested in forking my repo instead of @AltarBeastiful 's - or my repo might just be fine as it is?

is there any update regarding the icon from Alexander King? because the one I use for now might not be appropriate, and I would be glad to change it.

**Comment by fenryxo** [comment]
Hello @kixam. I've created a repository tiliado/nuvola-app-soundcloud. You can push there your local copy of your repository to remove any links with the original ipf/nuvola-soundcloud repository.

https://github.com/tiliado/nuvola-app-soundcloud

Then, you can work together with @SteVwonder and notify me once you are ready for code review.

> is there any update regarding the icon from Alexander King?

Alexander has time till December to prepare new icon set.

**Comment by SteVwonder** [comment]
I think your repo is probably good as is :smile:

**Comment by kixam** [comment]
I uploaded it to the above-mentioned location as a "fresh repo" even if I do not like this much. I did it anyway because there is not much left from the original repo I forked, and because my few commits were not much relevant nor numerous.

Before uploading, I amended the licence, readme, and metadata. I am new to Nuvola development, so feel free to change anything inappropriate.

**Comment by fenryxo** [comment]
@kixam, is your core ready for review then?

**Comment by kixam** [comment]
I think it is

**Comment by fenryxo** [comment]
Great, added to Roadmap for October 2015 tiliado/nuvolaplayer.project#13

**Comment by fenryxo** [comment]
Hello @kixam. I regret to tell you I don't expect to have time for the code review and packaging of your integration script until the release of the stable version of Nuvola Player 3.0, which is due at the end of December 2015. Since the funding of the Nuvola Player project has recently dropped significantly, I cannot spend time on tasks that are not crucial for the stable release. However, the release cycle of web app scripts is not bound to Nuvola Player releases, so I can prepare packages for your script in January 2016 regardless of the development progress and the release date of Nuvola Player 3.1. Thanks for your understanding and patience.

**Comment by fenryxo** [comment]
Hello @kixam. Nuvola Player 3.0.0 Stable is out and I'm going to get around to code reviews in February. There have been [some changes in Service Integration Guidelines and an introduction of Format requirements flags](https://groups.google.com/d/msg/nuvola-player-devel/Eec-MkJJ0cA/C-pyEsL4AQAJ). Could you update your script and ping me when you are done? Thanks!

**Comment by kixam** [comment]
Hello @fenryxo 

* I added a CONTRIBUTE.md file based on your template
* I added appropriate Format requirement flag
* I adjusted indentation, brackets, and braces, but some lines are still more than 100 characters (were not with previous indentation of 2 spaces)
* however, I still do not have a proper set of icons, but I understand I do not have any action to take on this, as the icons are to be provided by Alexander King?
